### PR TITLE
Unit test that simulate a 500 block reorg

### PR DESCRIPTION
**Pull request by dangershony**
This will also fix the failing tests we had til now

### CODE REVIEW

File: Stratis.Bitcoin.Features.Consensus/Checkpoints.cs [hunk 1]
```diff
@@ -138,6 +138,7 @@ public Checkpoints(Network network)
         {
             if (network.Equals(Network.Main)) this.checkpoints = bitcoinMainnetCheckpoints;
             else if (network.Equals(Network.TestNet)) this.checkpoints = bitcoinTestnetCheckpoints;
+            else if (network.Equals(Network.RegTest)) this.checkpoints = new Dictionary<int, CheckpointInfo>();
```
Comment (Aprogiena) [inline Stratis.Bitcoin.Features.Consensus/Checkpoints.cs]: I don't get it. I'm reading this over and over again and I don't get it :(
What am I missing? Why is this needed?

**Comment by Aprogiena** [comment]
Tried to run the test, but no luck for about 5 minutes (still running). How long is it supposed to run?

File: Stratis.Bitcoin.Features.Consensus/Checkpoints.cs (hunk 1 above)
Comment (dangershony) [reply Stratis.Bitcoin.Features.Consensus/Checkpoints.cs]: This is only for the truly knowledgeable in blockchain ;)

**Comment by dangershony** [comment]
Test can run for a while  as it needs to mine 500 blocks, maybe we can make the mining more efficient

File: Stratis.Bitcoin.Features.Consensus/Checkpoints.cs (hunk 1 above)
Comment (dangershony) [reply Stratis.Bitcoin.Features.Consensus/Checkpoints.cs]: Seriously, this is related to an issue in NBitcoin where POS sets static flags when initiating the stratis network.
I am working on fixing this.

**Comment by Aprogiena** [comment]
OK, it finished after 7 minutes, which is little long, but maybe OK. However, the test passed, which I don't like :-) I would like the test to fail actually until the CheckSync feature is implemented correctly.

File: Stratis.Bitcoin.Features.Consensus/Checkpoints.cs (hunk 1 above)
Comment (Aprogiena) [reply Stratis.Bitcoin.Features.Consensus/Checkpoints.cs]: I still don't understand this. Do you say that network.Equals gives TRUE for some of the IFs we've had there before in case the network was RegTest and not to go with the final else?

**Comment by dangershony** [comment]
We can maybe make it pass faster by making the reorg threshold configurable, we then set it to a lower number and mining less blocks to prevent the reorg

**Comment by Aprogiena** [comment]
Yes that is a good idea, we can test with 50 instead of 500.

File: Stratis.Bitcoin.Features.Consensus/Checkpoints.cs (hunk 1 above)
Comment (Aprogiena) [reply Stratis.Bitcoin.Features.Consensus/Checkpoints.cs]: OK, you've explained this to me. Weird!

**Comment by Aprogiena** [comment]
OK, so I've read the whole test now and I think I understand it +-. So yes, it actually tests the behavior that we would like to prevent - at the end this test ends up with all 3 guys synced on the same chain, but we would like two different forks to be there at the end and the nodes diverged forever exactly because the long reorg would be prevented. But fair enough, it gives me enough clues to start with on fixing PR 492.

So in general I'd like to ask you to modify the test accordingly:
- use `PosConsensusOptions.MaxReorgLength` and set it to some lower value, say 25 blocks to have this test finished in much reasonable time (within 30 seconds)
- generate those two forks as you do with length of something more than `PosConsensusOptions.MaxReorgLength`
- actually test that the tested node does not accept the longer chain from reorg node and that it considers itself as fully synced

**Comment by Aprogiena** [comment]
So, I've been testing this a little bit and the test does not use Stratis at all. It rather uses Bitcoin. `PosConsensusOptions` don't seem to be available and when the blocks are accepted, `PosConsensusValidator` is not used at all.

The long reorg prevention should only be implemented on Stratis only, so this test should reflect that. Currently, the test is not very useful, as I hoped, for the development of that feature because those nodes are not running on Stratis network.

**Comment by Aprogiena** [comment]
Also the PR's description should contain `Fix #493` because that is what it aims to do.

**Comment by dangershony** [comment]
Ah right the reorg prevention is the POS validator hmm..
We can simulate this, with the pow part of the pos validator, however it will be hard to run a stratis node,
Let me revisit this test and try it to use the POS network.
